From 0b043802dbfc550cc57d3537806d86cf9cdd376c Mon Sep 17 00:00:00 2001
From: Jakub Pawlowski <jpawlowski@google.com>
Date: Mon, 4 Apr 2022 21:49:42 +0200
Subject: [PATCH] GATT multi read: account for length for variable length read

The response will contain length, then the bytes with actual content.
Account for the 2 byte length field when computing how much data can be
squezed into a packet.

Bug: 227618988
Test: compilation
Change-Id: I6ddee3e5c50ce395fa4cc1b538d256a406a415f0
(cherry picked from commit 7056e665e7d4782af9474b1645c59afbb2b76efd)
Merged-In: I6ddee3e5c50ce395fa4cc1b538d256a406a415f0
---
 stack/gatt/gatt_sr.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/stack/gatt/gatt_sr.cc b/stack/gatt/gatt_sr.cc
index c0aedfafa..39c7125b7 100644
--- a/stack/gatt/gatt_sr.cc
+++ b/stack/gatt/gatt_sr.cc
@@ -172,6 +172,9 @@ static void build_read_multi_rsp(tGATT_SR_CMD* p_cmd, uint16_t mtu) {
 
     if (p_rsp != NULL) {
       total_len = (p_buf->len + p_rsp->attr_value.len);
+      if (p_cmd->multi_req.variable_len) {
+        total_len += 2;
+      }
 
       if (total_len > mtu) {
         /* just send the partial response for the overflow case */
-- 
2.36.0.550.gb090851708-goog

